libs/providers/CENNZApiProvider.tsx

```diff
@@ -19,19 +19,20 @@ const CENNZApiProvider: FC<CENNZApiProviderProps> = ({
 }) => {
 	const [api, setApi] = useState<Api>(null);
 
-	useEffect(() => {
-		const instance = new Api({
+	const initApi = async (endpoint: string) => {
+		const instance = await Api.create({
 			provider: endpoint,
 		});
 
-		instance.isReady.then(() => {
-			setApi(instance);
-			window.onunload = async () => await instance.disconnect();
-		});
+		window.onunload = async () => await instance.disconnect();
 
 		return () => {
 			void instance.disconnect();
 		};
+	};
+
+	useEffect(() => {
+		void initApi(endpoint);
 	}, [endpoint]);
```

Comment:
Couple of issues with this
- The `() => { void instance.disconnect();};` is unreachable for `useEffect`, so no clean up there when component is unmounted (unlikely given it's in the top level Provider, but good practice anyway)
- Unless it's a special requirement for React 18 but I don't see the need to declare `initApi` outside of `useEffect`. You can achieve the same thing by declaring `initApi` and triggering it all inside `useEffect`, at least that way it doesn't get created everything there is a render